### PR TITLE
Added string prepending.

### PR DESCRIPTION
**Pull request by elland**
Example usage:

```swift
path.prepend("/")
let normalizedPath = path.prepending("/")
```

### CODE REVIEW

File: Tests/StringTests.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class StringTests: XCTestCase {
+    let prefix = "prefix:"
+
+    func testPrepending() {
+        var mutableString = "a string that I can change"
+
+        XCTAssertFalse(mutableString.hasPrefix(prefix))
+        mutableString.prepend(prefix)
+        XCTAssertTrue(mutableString.hasPrefix(prefix))
```
Comment (designatednerd) [inline Tests/StringTests.swift]: In both of these tests, it's probably also worth testing the full string is what you expect it to be just to validate that you didn't do something silly like dropping the bit that you were trying to append

**Comment by elland** [comment]
Done :)

**Comment by elland** [comment]
Would be great if we could integrate this with CI.

File: Sources/String+Prepending.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+//
+
+import Foundation
+
+extension String {
+    public mutating func prepend(_ aString: String) {
+        self = aString.appending(self)
+    }
+
+    public func prepending(_ aString: String) -> String {
```
Comment (designatednerd) [inline Sources/String+Prepending.swift:22]: wanna add some docs on these?

**Comment by designatednerd** [comment]
It is integrated with CI, it just wasn't set to build PRs off forks. I've now enabled that so when you push your next changes, it should work.

**Comment by designatednerd** [comment]
(I also went through and made sure the other `Sweet*` repos can build forked PRs in the future)

**Comment by elland** [comment]
Nice. :) Any more changes?

**Comment by designatednerd** [comment]
@Elland requested more documentation plz

File: Sources/String+Prepending.swift (hunk 2 above)
Comment (elland) [reply Sources/String+Prepending.swift:22]: Did not see this comment before. UI playing tricks!

**Comment by elland** [comment]
CI was quick enough. :)

**Review by designatednerd — APPROVED** [review]

**Comment by designatednerd** [comment]
@Elland Bagged and tagged as `3.1.0`

**Pull request merged**